lib/active_merchant/billing/gateways/sum_up.rb

```diff
@@ -0,0 +1,192 @@
+          gsub(%r(("number\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r(("cvv\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]')
+      end
+
+      private
+
+      def upsert_checkout(money, payment, options = {})
+        if options[:checkout_id].present?
+          find_checkout(options[:checkout_id])
+        else
+          create_checkout(money, payment, options)
+        end
+      end
+
+      def find_checkout(checkout_id)
+        commit('checkouts/' + checkout_id, {}, :get)
+      end
+
+      def create_checkout(money, payment, options)
+        post = {}
+
+        add_merchant_data(post, options)
+        add_invoice(post, money, options)
+        add_address(post, payment, options)
+        add_customer_data(post, payment, options)
+
+        commit('checkouts', post)
+      end
+
+      def complete_checkout(checkout_id, payment, options = {})
+        post = {}
+
+        add_payment(post, payment, options)
+
+        commit('checkouts/' + checkout_id, post, :put)
+      end
+
+      def add_customer_data(post, payment, options)
+        post[:customer_id]            = options[:customer_id]
+        personal_details              = post[:personal_details] ||= {}
+        personal_details[:email]      = options[:email]
+        personal_details[:first_name] = payment&.first_name
+        personal_details[:last_name]  = payment&.last_name
+        personal_details[:tax_id]     = options[:tax_id]
+      end
+
+      def add_merchant_data(post, options)
+        # Required field: pay_to_email
+        # Description: Email address of the merchant (in this case Spreedly)
+        # to whom the payment is made.
+        post[:pay_to_email] = @options[:pay_to_email]
+        post[:redirect_url] = options[:redirect_url]
+        post[:return_url]   = options[:return_url]
+      end
+
+      def add_address(post, creditcard, options)
+        personal_details      = post[:personal_details] ||= {}
+        address               = personal_details[:address] = {}
+        address[:city]        = options[:billing_address][:city]
+        address[:state]       = options[:billing_address][:state]
+        address[:country]     = options[:billing_address][:country]
+        address[:line_1]      = options[:billing_address][:address1]
+        address[:postal_code] = options[:billing_address][:zip]
+      end
+
+      def add_invoice(post, money, options)
+        post[:checkout_reference] = options[:checkout_reference]
+        post[:amount]             = amount(money)
+        post[:currency]           = (options[:currency] || currency(money))
+        post[:description]        = options[:description]
+      end
+
+      def add_payment(post, payment, options)
+        post[:payment_type] = options[:payment_type] if PAYMENT_TYPES.include?(options[:payment_type])
+        card                = post[:card] = {}
+        card[:name]         = [payment.first_name, payment.last_name].join(' ')
```

Comment:
💬 Comment:

Maybe just using the `ActiveMerchant::Billing::CreditCard#name` method?